### PR TITLE
fix: eng 1091 upload video error

### PR DESCRIPTION
**Pull request by scottkelso**
# Description

### Issue

If an upload video error occurs but the user tries again, there error text still remains as well as the uploading/processing text

[Link to Linear Todo](https://linear.app/jesus-film-project/issue/ENG-1091/upload-video-error)

### Solution

Set error state to false onDrop

# External Changes

Refactored test class

# How to test
When a video upload error occurs (maybe forced by network change eta), the user should be able to retry again.
On another attempt you should never have both the old error text and the new uploading/processing text together.

### CODE REVIEW

File: apps/journeys-admin/src/components/Editor/Slider/Settings/Drawer/VideoLibrary/VideoFromCloudflare/AddByFile/AddByFile.tsx [hunk 1]
```diff
@@ -100,6 +100,7 @@ export function AddByFile({
     settooManyFiles(false)
     setfileInvalidType(false)
     setfileRejected(false)
+    setError(undefined)
```
Comment (csiyang) [inline apps/journeys-admin/src/components/Editor/Slider/Settings/Drawer/VideoLibrary/VideoFromCloudflare/AddByFile/AddByFile.tsx:103]: anyway to unit test this?

**Review by csiyang — CHANGES_REQUESTED** [review]

**Review by jianwei1 — COMMENTED** [review]
Great job on extracting the mocks! The spec file looks much cleaner now

File: apps/journeys-admin/src/components/Editor/Slider/Settings/Drawer/VideoLibrary/VideoFromCloudflare/AddByFile/AddByFile.tsx (hunk 1 above)
Comment (scottkelso) [reply apps/journeys-admin/src/components/Editor/Slider/Settings/Drawer/VideoLibrary/VideoFromCloudflare/AddByFile/AddByFile.tsx:103]: Thought of a way thismorning. Have a look!

**Review by csiyang — APPROVED** [review]

File: apps/journeys-admin/src/components/Editor/Slider/Settings/Drawer/VideoLibrary/VideoFromCloudflare/AddByFile/AddByFile.spec.tsx [hunk 2]
```diff
@@ -1,233 +1,177 @@
...
-      value: [file]
-    })
-    fireEvent.drop(input)
-    await waitFor(() => expect(getByText('Uploading...')).toBeInTheDocument())
-    expect(getByTestId('Upload1Icon')).toBeInTheDocument()
+    await dropTestVideo()
+    await completeUpload(testStack)
+    await waitFor(() => expect(onChange).toHaveBeenCalledWith('uploadId'))
+    await waitFor(() =>
+      expect(screen.queryByText('Processing...')).not.toBeInTheDocument()
```
Comment (csiyang) [inline apps/journeys-admin/src/components/Editor/Slider/Settings/Drawer/VideoLibrary/VideoFromCloudflare/AddByFile/AddByFile.spec.tsx:136]: is it possible to check this text is in the document before its completed?

Comment (scottkelso) [reply apps/journeys-admin/src/components/Editor/Slider/Settings/Drawer/VideoLibrary/VideoFromCloudflare/AddByFile/AddByFile.spec.tsx:136]: That's what `should complete a file upload and call onChange` test is doing.

**Review by csiyang — APPROVED** [review]

**Pull request merged**